File: apps/rails/spec/factories/wise_recipients.rb
# frozen_string_literal: true

FactoryBot.define do
  factory :wise_recipient do
    user
    wise_credential

    country_code { "US" }
    currency { "USD" }
    recipient_id { "148563324" } # Test Wise Sandbox Recipient ID
    last_four_digits { "1234" }
    account_holder_name { "Jane Q. Contractor" }
  end
end

File: e2e/factories/wiseRecipients.ts
import { db } from "@test/db";
import { usersFactory } from "@test/factories/users";
import { wiseRecipients } from "@/db/schema";
import { assert } from "@/utils/assert";

export const wiseRecipientsFactory = {
  create: async (overrides: Partial<typeof wiseRecipients.$inferInsert> = {}) => {
    const [wiseRecipient] = await db
      .insert(wiseRecipients)
      .values({
        userId: overrides.userId || (await usersFactory.create()).user.id,
        wiseCredentialId: 1n,
        countryCode: "US",
        currency: "USD",
        recipientId: "148563324", // Test Wise Sandbox Recipient ID
        lastFourDigits: "1234",
        accountHolderName: "Jane Q. Contractor",
        usedForInvoices: true,
        ...overrides,
      })
      .returning();
    assert(wiseRecipient != null);

    return { wiseRecipient };
  },
};
